### PR TITLE
Move keyboard USB IDs and strings to data driven: K

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

Previous PR: #17837 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keycapsss/o4l_5x12/info.json [hunk 1]
```diff
@@ -1,7 +1,13 @@
 {
-    "keyboard_name": "O4L 5x12",
+    "keyboard_name": "O4L:5x12",
```
Comment (noroadsleft) [inline keyboards/keycapsss/o4l_5x12/info.json:2]: Is this colon supposed to be here?

File: keyboards/kudox/rev1/info.json [hunk 2]
```diff
@@ -1,7 +1,13 @@
 {
+    "keyboard_name": "The Kudox Keyboard",
```
Comment (noroadsleft) [inline keyboards/kudox/rev1/info.json:2]:
```suggestion
    "keyboard_name": "The Kudox Keyboard Rev1",
```

File: keyboards/kudox/rev2/info.json [hunk 3]
```diff
@@ -1,7 +1,13 @@
 {
+    "keyboard_name": "The Kudox Keyboard",
```
Comment (noroadsleft) [inline keyboards/kudox/rev2/info.json:2]:
```suggestion
    "keyboard_name": "The Kudox Keyboard Rev2",
```

File: keyboards/kudox/rev3/info.json [hunk 4]
```diff
@@ -1,7 +1,13 @@
 {
+    "keyboard_name": "The Kudox Keyboard",
```
Comment (noroadsleft) [inline keyboards/kudox/rev3/info.json:2]:
```suggestion
    "keyboard_name": "The Kudox Keyboard Rev3",
```

File: keyboards/kudox_full/rev1/info.json [hunk 5]
```diff
@@ -1,7 +1,13 @@
 {
+    "keyboard_name": "The Kudox Full Keyboard",
```
Comment (noroadsleft) [inline keyboards/kudox_full/rev1/info.json:2]:
```suggestion
    "keyboard_name": "The Kudox Full Keyboard Rev1",
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Some general notes:

- Some of the KBDfans boards use `"manufacturer": "KBDfans"`, and others are set to `"manufacturer": "KBDFans"` with a capital F.
- Most boards with multiple revisions are now set to specify the revision in `"keyboard_name"`, which I think is a good change - but the Suihankey by kakunpc does not. Is that deliberate?
- `keyboards/keyboardio/model01/config.h` still has the USB strings; they haven't been removed in the conversion to data-driven config.

File: keyboards/keycapsss/o4l_5x12/info.json (hunk 1 above)
Comment (fauxpark) [reply keyboards/keycapsss/o4l_5x12/info.json:2]: Yes.
image

**Comment by fauxpark** [comment]
> Some of the KBDfans boards use "manufacturer": "KBDfans", and others are set to "manufacturer": "KBDFans" with a capital F.

> Most boards with multiple revisions are now set to specify the revision in "keyboard_name", which I think is a good change - but the Suihankey by kakunpc does not. Is that deliberate?

I tried to keep whatever was there originally, and made tweaks as I thought necessary. I'm not too bothered by it; the only thing I don't touch is the VID/PID as that will break VIA.

File: keyboards/keycapsss/o4l_5x12/info.json (hunk 1 above)
Comment (noroadsleft) [reply keyboards/keycapsss/o4l_5x12/info.json:2]: Ah, I misread the `config.h` changes.

**Review by noroadsleft — APPROVED** [review]

**Comment by spidey3** [comment]
I looks like some of the kdbfans boards were not done (e.g. kbdfans/kbd75). Was this intentional?

**Comment by fauxpark** [comment]
Yes: https://github.com/qmk/qmk_firmware/pull/16481

**Review by keyboard-magpie — APPROVED** [review]

**Pull request merged**